layersvt/vulkan_json_layer.hpp

```diff
@@ -783,7 +793,7 @@ class PipelineData {
     void dumpObjectReservationInfo()
     {
         std::string filename = getenv("VK_JSON_FILE_PATH");
-        filename += getProcessName();
+        filename = getBaseDirectoryPath(filename) + getProcessName();
```

Comment:
I was thinking getBaseDirectoryPath would also include the `getenv("VK_JSON_FILE_PATH")`  part (or the version from line 570 - whichever is better).

Reply:
Ah got it, I've gone ahead and moved `getenv("VK_JSON_FILE_PATH")` as well into getBaseDirectoryPath.

Also moved the invocation of `getBaseDirectoryPath()` to the ctor of `PipelineData` to avoid invoking it multiple times. Did the same with `getProcessName` as well since this was being invoked multiple times as well.